File: Medium/Python 3/Rotate_Array(189).py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        i = 0
        while i < n-(k % n):
            v = nums.pop(0)
            nums.append(v)
            i += 1
        return nums

File: Medium/Java/Rotate_Array(189).java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k%nums.length; // if k > nums.length
        
        // reverse 1st half
        reverse(nums, 0, nums.length-k-1);
        
        // reverse 2nd half
        reverse(nums, nums.length-k, nums.length-1);
        
        // reverse whole array
        reverse(nums, 0, nums.length-1);
    }
    
    public void reverse(int[] nums, int i, int j){
        int li = i;
        int ri = j;
        
        while(li < ri){
            int temp = nums[li];
            nums[li] = nums[ri];
            nums[ri] = temp;
            
            li++;
            ri--;
        }
    }
}